Microsoft.Toolkit/Diagnostics/Generated/Guard.Comparable.Numeric.tt

```diff
@@ -0,0 +1,237 @@
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="value"/> is &lt;= <paramref name="minimum"/>.</exception>
+        /// <remarks>The method is generic to avoid boxing the parameters, if they are value types.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsGreaterThan(<#=type#> value, <#=type#> minimum, string name)
+        {
+            if (value <= minimum)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForIsGreaterThan(value, minimum, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input value must be greater than or equal to a specified value.
+        /// </summary>
+        /// <param name="value">The input <see cref="<#=type#>"/> value to test.</param>
+        /// <param name="minimum">The inclusive minimum <see cref="<#=type#>"/> value that is accepted.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="value"/> is &lt; <paramref name="minimum"/>.</exception>
+        /// <remarks>The method is generic to avoid boxing the parameters, if they are value types.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsGreaterThanOrEqualTo(<#=type#> value, <#=type#> minimum, string name)
+        {
+            if (value < minimum)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForIsGreaterThanOrEqualTo(value, minimum, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input value must be in a given range.
+        /// </summary>
+        /// <param name="value">The input <see cref="<#=type#>"/> value to test.</param>
+        /// <param name="minimum">The inclusive minimum <see cref="<#=type#>"/> value that is accepted.</param>
+        /// <param name="maximum">The exclusive maximum <see cref="<#=type#>"/> value that is accepted.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="value"/> is &lt; <paramref name="minimum"/> or >= <paramref name="maximum"/>.</exception>
+        /// <remarks>The method is generic to avoid boxing the parameters, if they are value types.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsInRange(<#=type#> value, <#=type#> minimum, <#=type#> maximum, string name)
+        {
+            if (value < minimum || value >= maximum)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForIsInRange(value, minimum, maximum, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input value must not be in a given range.
```

Comment:
Ah, I see the subtle differences here between InRange and InBetween... wondering how to make this easier to process, is there an existing .NET equivalent to point me to for a reference?

Reply:
Mh I wouldn't know of an existing .NET API that's equivalent to that I'm afraid 🤔

I guess maybe we could mention the mathematical range in the remarks for each API?

- `IsInRange` ---> **[min, max)**
- `IsBetween` ---> **(min, max)**
- `IsBetweenOrEqualTo` ---> **[min, max]**

Reply:
Yeah, I know not everyone may be familiar with that notation, but at a glance, especially in the breakdown overview doc it'd make it quickly easy to pick-up the differences.

Reply:
Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3131/commits/67fcd478ab4787eb4d57e67019059476b7d2e40b and https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/pull/302/commits/61452b3c9d5538c6683f11aa9cf66b7de32c5d9f 👍